pkg/cephfs/util.go

```diff
@@ -114,3 +114,21 @@ func (cs *ControllerServer) validateDeleteVolumeRequest() error {
 
 	return nil
 }
+
+// Controller expand volume request validation
+func (cs *ControllerServer) validateExpandVolumeRequest(req *csi.ControllerExpandVolumeRequest) error {
+	if err := cs.Driver.ValidateControllerServiceRequest(csi.ControllerServiceCapability_RPC_EXPAND_VOLUME); err != nil {
+		return fmt.Errorf("invalid ExpandVolumeRequest: %v", err)
+	}
+
+	if req.GetVolumeId() == "" {
+		return status.Error(codes.InvalidArgument, "Volume ID cannot be empty")
+	}
+
+	capRange := req.GetCapacityRange()
+	if capRange == nil {
+		return status.Error(codes.InvalidArgument, "CapacityRange cannot be empty")
```

Comment:
How do you unset the quota for no size limit? See the `create_subvolume` definition below.

```
    def create_subvolume(self, spec, size=None, namespace_isolated=True, mode=0o755, pool=None):                                                                                              
        """                                                                                                                                                                                   
        Set up metadata, pools and auth for a subvolume.                                                                                                                                      
                                                                                                                                                                                              
        This function is idempotent.  It is safe to call this again                                                                                                                           
        for an already-created subvolume, even if it is in use.                                                                                                                               
                                                                                                                                                                                              
        :param spec: subvolume path specification                                                                                                                                             
        :param size: In bytes, or None for no size limit       
```

Reply:
@joscollin you will never get a request with size NONE from CO, so we are good .

Reply:
What I meant was, You need to allow size `NONE` (or what ever) to allow the user to unset the quota.